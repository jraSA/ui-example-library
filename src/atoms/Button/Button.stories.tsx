import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import Button from '.';

export default {
    title: 'Atoms/Button',
    component: Button,
} as Meta;

const Template = () => <Button>My text</Button>;

export const SimpleButtonComponent = Template.bind({});
