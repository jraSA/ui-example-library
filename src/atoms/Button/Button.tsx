import styled from 'styled-components';
import BaseTheme from '../../theme/BaseTheme';
import { getThemeValue } from '../../theme/ThemeUtils';

const { margin } = BaseTheme;

const { color, fonts, padding, radii, space } = BaseTheme;

const { black, white, lightboxBackground, grays } = color;
const { primaryFont, fontSizes } = fonts;

const Button = styled.button`
    width: 20%;
    margin-top: ${margin[5]}px;
    margin-bottom: ${margin[5]}px;
    padding: ${padding[4]}px;
    border-radius: ${radii[3]}px;

    align-self: center;

    background-color: ${(props) => getThemeValue(props, 'primaryBackground', white)};
    box-shadow: ${space[1]}px ${space[1]}px ${space[2]}px ${lightboxBackground};
    &:hover {
        box-shadow: ${space[2]}px ${space[2]}px ${space[3]}px ${lightboxBackground};
        background-color: ${(props) => getThemeValue(props, 'secondaryBackground', grays[300])};
    };
    outline: none;
    border: none;

    font-family: ${primaryFont}, arial, sans-serif;
    font-size: ${(props) => getThemeValue(props, 'titleSize', fontSizes[8])}px;    
    color: ${(props) => getThemeValue(props, 'titleColor', black)};
`;

export default Button;