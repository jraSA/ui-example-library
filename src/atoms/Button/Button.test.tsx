import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';

describe('Render test', () => {
    const wrapper = renderer.create(<Button>foo-boo</Button>);

    test('WHEN I render the component THEN it should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
