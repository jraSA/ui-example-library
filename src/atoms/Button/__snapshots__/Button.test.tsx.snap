// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render test WHEN I render the component THEN it should match the snapshot 1`] = `
<button
  className="sc-bdfBwQ llOwxJ"
>
  foo-boo
</button>
`;
